importer/NoNo.py

```diff
@@ -1,53 +1,102 @@
+        desc_norwegian = base_norwegian.format(category, municip)
+        self.add_description("nb", desc_norwegian)
+
+        base_english = "cultural property in {}, Norway"
+        desc_english = base_english.format(municip)
+        self.add_description("en", desc_english)
 
-        NOTE
-        Some of these are in all caps or have multiple spaces:
-        UTSIRA FYRSTASJON
-        SØGARD FJONE  -  FJONE SØNDRE
-        VÅLE PRESTEGÅRD, museum
+        base_swedish = "kulturarv i {}, Norge"
+        desc_swedish = base_swedish.format(municip)
+        self.add_description("sv", desc_swedish)
 
-        TODO
-        *Normalize - to title case?
-            It contains some old-style numbers, and these will be broken:
-            XXVIII -> Xxviii
+    def update_labels(self):
+        """
+        Add label in Bokmål.
 
-        * rm extra whitespace
+        Some of self.navn are in all caps.
+        These are normalized:
+        FÆRDER FYRSTASJON -> Færder Fyrstasjon
         """
-        for part in self.navn.split(" "):
-            if part.isupper():
-                print(part, "----", part.capitalize())
+        name = self.navn
+        if name.isupper():
+            name = name.title()
+        self.add_label("nb", name)
 
     def set_no(self):
-        self.add_statement("norwegian_monument_id", self.id)
+        """Add the Norwegian monument ID P758."""
+        self.add_statement("norwegian_monument_id", str(self.id))
+
+    def get_municip_name(self):
+        """Get the value of self.kommune without brackets."""
+        return utils.remove_markup(self.kommune)
+
+    def set_adm_location(self):
+        """Set the adm location based on kommunenummer and offline file."""
+        all_codes = self.data_files["municipalities"]
+        if self.has_non_empty_attribute("kommunenr"):
+            municip_code = str(self.kommunenr)
+            if len(municip_code) == 3:
+                municip_code = "0" + municip_code
+            match = [x["item"]
+                     for x in all_codes if x["municipNumber"] == municip_code]
+            if len(match) == 0:
+                self.add_to_report("kommunenr",
+                                   municip_code, "located_adm")
+            else:
+                self.add_statement("located_adm", match[0])
 
-    def __init__(self, db_row_dict, mapping, data_files, existing):
-        Monument.__init__(self, db_row_dict, mapping, data_files, existing)
+    def set_special_type(self):
+        """
+        Set special 'is' based on self.kategori and online mapping.
+
+        https://www.wikidata.org/wiki/Wikidata:WikiProject_WLM/Mapping_tables/no_(no)/categories
+        """
+        glossary = self.data_files["categories"]["mappings"]
+        category = self.kategori.lower()
+        try:
+            special_type = [glossary[x]["items"]
+                            for x in glossary
+                            if x.lower() == category][0][0]
```

Comment:
separate list comprehension from getting to the contents (i.e. the `[0][0]`).

See also https://github.com/Vesihiisi/COH-tools/pull/64/files#r124743473